src/helics/common/TomlProcessingFunctions.cpp

```diff
@@ -28,68 +28,56 @@ toml::Value loadToml(const std::string& tomlString)
             // just pass through this was an assumption
         }
     }
-    std::ifstream file(tomlString);
+    std::ifstream file(tomlString, std::ios_base::binary);
 
     if (file.is_open()) {
-        toml::ParseResult pr = toml::parse(file);
-        if (!pr.valid()) {
-            throw(std::invalid_argument(pr.errorReason));
-        }
-
-        return pr.value;
+        return toml::parse(file);
     }
     return loadTomlStr(tomlString);
 }
 
-toml::Value loadTomlStr(const std::string& tomlString)
+toml::value loadTomlStr(const std::string& tomlString)
 {
     std::istringstream tstring(tomlString);
-    toml::ParseResult pr = toml::parse(tstring);
-    if (pr.valid()) {
-        return pr.value;
-    }
-    throw(std::invalid_argument(pr.errorReason));
+    toml::value pr = toml::parse(tstring);
+    return pr;
 }
 
+static const std::string emptyString;
+
 /** read a time from a JSON value element*/
-helics::Time loadTomlTime(const toml::Value& timeElement, time_units defaultUnits)
+helics::Time loadTomlTime(const toml::value& timeElement, time_units defaultUnits)
 {
-    if (timeElement.is<toml::Table>()) {
-        auto units = timeElement.find("units");
-        if (units != nullptr) {
-            defaultUnits = gmlc::utilities::timeUnitsFromString(units->as<std::string>());
+    if (timeElement.is_table()) {
+        auto& units = toml::find_or<std::string>(timeElement, "units", emptyString);
+        if (!units.empty()) {
+            defaultUnits = gmlc::utilities::timeUnitsFromString(units);
         }
-        auto val = timeElement.find("value");
-        if (val != nullptr) {
-            if (val->is<int64_t>()) {
-                return {val->as<int64_t>(), defaultUnits};
+        toml::value emptyVal;
+        auto val = toml::find_or(timeElement, "value", emptyVal);
+        if (!val.is_uninitialized()) {
+            if (val.is_integer()) {
+                return {val.as_integer(), defaultUnits};
             }
-            return {val->as<double>() * toSecondMultiplier(defaultUnits)};
+            return {val.as_floating() * toSecondMultiplier(defaultUnits)};
```

Comment:
Should there be a check here to make sure the value is a floating point number?

Reply:
if it isn't that is an incorrectly specified file, but the same was true before.

Reply:
Yea, I noticed a few other places that had checks. I don't know if a linting tool for input files is needed to catch issues with fields that are given unexpected types of values.